File: GTTS/INL.TrainingService.Models/GetTrainingEvent_Item.cs
using System;
using System.Collections.Generic;

namespace INL.TrainingService.Models
{
    public class GetTrainingEvent_Item : IGetTrainingEvent_Item
    {
        public long TrainingEventID { get; set; }
        public string Name { get; set; }
        public string NameInLocalLang { get; set; }
        public int TrainingEventTypeID { get; set; }
        public string TrainingEventTypeName { get; set; }
        public string Justification { get; set; }
        public string Objectives { get; set; }
        public string ParticipantProfile { get; set; }
        public string SpecialRequirements { get; set; }
        public int? ProgramID { get; set; }
        public int? KeyActivityID { get; set; }
        public string KeyActivityName { get; set; }
        public long? TrainingUnitID { get; set; }
        public string BusinessUnitAcronym { get; set; }
        public string BusinessUnitName { get; set; }
        public int? CountryID { get; set; }
        public int? PostID { get; set; }
        public int? ConsularDistrictID { get; set; }
        public int? OrganizerAppUserID { get; set; }
        public int? PlannedParticipantCnt { get; set; }
        public int? PlannedMissionDirectHireCnt { get; set; }
        public int? PlannedNonMissionDirectHireCnt { get; set; }
        public int? PlannedMissionOutsourceCnt { get; set; }
        public int? PlannedOtherCnt { get; set; }
        public decimal? EstimatedBudget { get; set; }
        public decimal? ActualBudget { get; set; }
        public int? EstimatedStudents { get; set; }
        public int? FundingSourceID { get; set; }
        public int? AuthorizingLawID { get; set; }
        public string Comments { get; set; }
        public int? ModifiedByAppUserID { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public DateTime? TravelStartDate { get; set; }
        public DateTime? TravelEndDate { get; set; }
        public DateTime? EventStartDate { get; set; }
        public DateTime? EventEndDate { get; set; }
        public List<ParticipantCount_Item> ParticipantCounts { get; set; }
        public int? TrainingEventStatusID { get; set; }
        public string TrainingEventStatus { get; set; }
        public GetTrainingEventAppUser_Item Organizer { get; set; }
        public GetTrainingEventAppUser_Item ModifiedBy { get; set; }

        public List<GetTrainingEventLocation_Item> TrainingEventLocations { get; set; }
        public List<GetTrainingEventUSPartnerAgency_Item> TrainingEventUSPartnerAgencies { get; set; }
        public List<GetTrainingEventProjectCode_Item> TrainingEventProjectCodes { get; set; }
        public List<GetTrainingEventStakeholder_Item> TrainingEventStakeholders { get; set; }
        public List<GetTrainingEventAttachment_Item> TrainingEventAttachments { get; set; }
        public List<GetTrainingCourseDefinitionProgram_Item> TrainingEventCourseDefinitionPrograms { get; set; }
        public List<GetTrainingEventKeyActivity_Item> TrainingEventKeyActivities { get; set; }
        public List<IAA_Item> IAAs { get; set; }
    }
}

File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.TrainingService.Models/get-training-event_item.ts


import { ParticipantCount_Item } from './participant-count_item';
import { GetTrainingEventAppUser_Item } from './get-training-event-app-user_item';
import { GetTrainingEventLocation_Item } from './get-training-event-location_item';
import { GetTrainingEventUSPartnerAgency_Item } from './get-training-event-uspartner-agency_item';
import { GetTrainingEventProjectCode_Item } from './get-training-event-project-code_item';
import { GetTrainingEventStakeholder_Item } from './get-training-event-stakeholder_item';
import { GetTrainingEventAttachment_Item } from './get-training-event-attachment_item';
import { GetTrainingCourseDefinitionProgram_Item } from './get-training-course-definition-program_item';
import { GetTrainingEventKeyActivity_Item } from './get-training-event-key-activity_item';
import { IAA_Item } from './iaa_item';
import { IGetTrainingEvent_Item } from './iget-training-event_item';

export class GetTrainingEvent_Item implements IGetTrainingEvent_Item {
  
	public TrainingEventID: number = 0;
	public Name: string = "";
	public NameInLocalLang: string = "";
	public TrainingEventTypeID: number = 0;
	public TrainingEventTypeName: string = "";
	public Justification: string = "";
	public Objectives: string = "";
	public ParticipantProfile: string = "";
	public SpecialRequirements: string = "";
	public ProgramID?: number;
	public KeyActivityID?: number;
	public KeyActivityName: string = "";
	public TrainingUnitID?: number;
	public BusinessUnitAcronym: string = "";
	public BusinessUnitName: string = "";
	public CountryID?: number;
	public PostID?: number;
	public ConsularDistrictID?: number;
	public OrganizerAppUserID?: number;
	public PlannedParticipantCnt?: number;
	public PlannedMissionDirectHireCnt?: number;
	public PlannedNonMissionDirectHireCnt?: number;
	public PlannedMissionOutsourceCnt?: number;
	public PlannedOtherCnt?: number;
	public EstimatedBudget?: number;
	public ActualBudget?: number;
	public EstimatedStudents?: number;
	public FundingSourceID?: number;
	public AuthorizingLawID?: number;
	public Comments: string = "";
	public ModifiedByAppUserID?: number;
	public ModifiedDate?: Date;
	public TravelStartDate?: Date;
	public TravelEndDate?: Date;
	public EventStartDate?: Date;
	public EventEndDate?: Date;
	public ParticipantCounts?: ParticipantCount_Item[];
	public TrainingEventStatusID?: number;
	public TrainingEventStatus: string = "";
	public Organizer?: GetTrainingEventAppUser_Item;
	public ModifiedBy?: GetTrainingEventAppUser_Item;
	public TrainingEventLocations?: GetTrainingEventLocation_Item[];
	public TrainingEventUSPartnerAgencies?: GetTrainingEventUSPartnerAgency_Item[];
	public TrainingEventProjectCodes?: GetTrainingEventProjectCode_Item[];
	public TrainingEventStakeholders?: GetTrainingEventStakeholder_Item[];
	public TrainingEventAttachments?: GetTrainingEventAttachment_Item[];
	public TrainingEventCourseDefinitionPrograms?: GetTrainingCourseDefinitionProgram_Item[];
	public TrainingEventKeyActivities?: GetTrainingEventKeyActivity_Item[];
	public IAAs?: IAA_Item[];
  
}






